File: week2_algorithmic_warmup/1_fibonacci_number/Fibonacci.java
import java.util.Scanner;

public class Fibonacci {
    private static long calc_fib(int n) {
        int[] fib_num = {0, 1};
        if (fib_num.length == n) {
            fib_num = append(fib_num, (fib_num[n - 1] + fib_num[n - 2]));
        }
        if (fib_num.length > n) {
            return fib_num[n];
        }
        return fib_num[n];
    }

    public static void main(String args[]) {
        Scanner in = new Scanner(System.in);
        int n = in.nextInt();
        int[] fib_table = {0, 1};
        if (n < fib_table.length) {
            for (int i = 2; i < n; i++) {

            }
        }
        System.out.println(calc_fib(n));
    }

    private static int[] append(int arr[], int x) {
        int i;
        int n = arr.length;
        // create a new array of size n+1
        int newarr[] = new int[n + 1];

        // insert the elements from
        // the old array into the new array
        // insert all elements till n
        // then insert x at n+1
        for (i = 0; i < n; i++)
            newarr[i] = arr[i];

        newarr[n] = x;

        return newarr;
    }
}

File: week2_algorithmic_warmup/1_fibonacci_number/Fibonacci.kt
import java.util.*

fun calcFibSlow(n: Long): Long {
    return if (n <= 1) n else calcFibSlow(n - 1) + calcFibSlow(n - 2)
}

fun calcFib123(n: Long): Long {
    if (n <= 1) {
        return n
    }
    var prev2: Long = 0
    var prev1: Long = 1
    var current: Long = 0
    for (i in 1..n) {
        current = prev2 + prev1
        prev1 = prev2
        prev2 = current
    }
    return current;
}

fun main(args: Array<String>) {
    val scanner = Scanner(System.`in`)
    val n = scanner.nextLong()
 //   println(calcFib(n))
}